library/timing.c

```diff
@@ -19,6 +19,18 @@
  *  This file is part of mbed TLS (https://tls.mbed.org)
  */
 
+/*
+ * for clock_gettime() from <time.h>
+ */
+#define _POSIX_C_SOURCE 200112L
```

Comment:
[FreeBSD doesn't define `gettimeofday()` if only `_POSIX_C_SOURCE `is defined, without `_XOPEN_SOURCE`.](https://github.com/freebsd/freebsd/blob/release/11.1.0/sys/sys/time.h#L494)

See relevant failed build log [here](https://gist.github.com/bensze01/870cb75042140eb235962beb590dd6cf) (please ignore the warnings about `static_assert`).

`_XOPEN_SOURCE >= 600` also [implies `_POSIX_C_SOURCE 200112L`](https://man7.org/linux/man-pages/man7/feature_test_macros.7.html).
```suggestion
/*
 * for clock_gettime() from <time.h>
 * and for gettimeofday() from <sys/time.h> on FreeBSD
 */
#define _XOPEN_SOURCE 600
```